scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,523 @@
+    end
+
+    -- middle blue door
+    if
+    mob:checkDistance(middleBlueDoorCoord.x, middleBlueDoorCoord.y, middleBlueDoorCoord.z) <= 5 then
+        if
+            pos.x <= -81 and
+            pos.y <= 1 and
+            pos.z >= 459 and
+            getLastDoor == 1
+        then
+            if
+                middleBlueDoor and
+                middleBlueDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(middleBlueToFirstBlue, xi.path.flag.COORDS)
+            elseif
+                middleBlueDoor and
+                middleBlueDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(middleBlueToMiddleYellow, xi.path.flag.COORDS)
+                mob:setLocalVar('[Zipacna]lastDoor', 2)
+            end
+        elseif
+            pos.x >= -76 and
+            pos.y <= 1 and
+            pos.z >= 460 and
+            getLastDoor >= 2
+        then
+            if
+                middleBlueDoor and
+                middleBlueDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(middleBlueToMiddleYellow, xi.path.flag.COORDS)
+            elseif
+                middleBlueDoor and
+                middleBlueDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(middleBlueToFirstBlue, xi.path.flag.COORDS)
+            end
+        end
+    end
+    -- middle yellow door
+    if
+    mob:checkDistance(middleYellowDoorCoord.x, middleYellowDoorCoord.y, middleYellowDoorCoord.z) <= 5 then
+        if
+            pos.x <= 78 and
+            pos.y <= 1 and
+            pos.z <= 461 and
+            getLastDoor <= 3
+        then
+            print ("12check distance to middle yellow door is a success! from blue middle door")
+            if
+                middleYellowDoor and
+                middleYellowDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(middleYellowToMiddleBlue, xi.path.flag.COORDS)
+            elseif
+                middleYellowDoor and
+                middleYellowDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(middleYellowToFirstYellow, xi.path.flag.COORDS)
+                mob:setLocalVar('[Zipacna]lastDoor', 3)
+            end
+        elseif
+            pos.x >= 81 and
+            pos.y <= 1 and
+            pos.z >= 459 and
+            getLastDoor == 4
+        then
+            if
+                middleYellowDoor and
+                middleYellowDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(middleYellowToFirstYellow, xi.path.flag.COORDS)
+            elseif
+                middleYellowDoor and
+                middleYellowDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(middleYellowToMiddleBlue, xi.path.flag.COORDS)
+                mob:setLocalVar('[Zipacna]lastDoor', 3)
```

Comment:
Instead of setting these, would it work to use `not entity:isFollowingPath()` to make sure you aren't constantly resetting the path with `pathThrough()`?

Reply:
Helping Hook out with this - So will comment on some of this on his behalf.
Since these have been extracted to a helper function, and `pathThrough()` is only getting called when in the presence of a door now, I am not sure that we need to use `isFollowingPath()` as the paths now end at each door where the check is performed to find a new path.

Might need to check the usage in the updated ammended commit to see if there is still a need to swap in some `isFollowingPath()` logic.